### DISCUSSION TITLE
Alias level PGP encryption

### DISCUSSION THREAD

**Discussion by leshik** [simple-login/app#1283] [Feature Requests]
I would love to have an option to add a PGP key to an alias so that only mail sent to this particular alias is encrypted before being forwarded to the mailbox.

The use case is to have only certain mail (e.g. banks, crypto exchanges, etc.) be encrypted, while the other is left as is so it could be read by ordinary email clients. In the case of Gmail, such a flow can be achieved already by creating an additional plus sign mailbox and setting a PGP key for it, then all mail ends up in the Gmail inbox where encrypted messages could be read in the Gmail web interface using e.g. Mailvelope browser extension. Other email providers often don't have such a hack, thus it would be great to set the PGP key per alias.

The main obstacle here is how to deal with the situation when the key is set for both an alias and a mailbox. The possible solution might be some priority mechanism, e.g. to use an alias key if it's set and then ignore the mailbox one in this case.

Ref https://github.com/simple-login/app/discussions/468

**Comment by nguyenkims** [comment]
@leshik We currently have a way to disable PGP on an alias, I wonder if that could work for you? Concretely, you can 
- add a PGP key for your mailbox: all emails sent to mailbox's aliases are encrypted with this key
- for aliases that don't need encryption, you can disable encryption.

This seems to be the inverse of what you want to do but maybe can work as a workaround in the meantime.

**Comment by leshik** [comment]
@nguyenkims Interesting. Actually, for now, I solved it by adding an additional "plus sign" mailbox with the PGP key, as I use Gmail. Your proposal should also work, but it all depends on what kind of mail prevails in one's mailbox: encrypted or unencrypted. Mine is unencrypted mainly, and I only want certain mail to be encrypted, thus it would be a pain to disable PGP encryption for every newly created alias.
